### DISCUSSION TITLE
Let, Stm, and the Stack

### DISCUSSION THREAD

**Discussion by magnus-madsen** [flix/flix#9165] [Ideas]
What do you think about chaining the definitions of `Let` and `Stm` to decrease stack usage?

```scala
case class LetMatch(bindings: List(Pattern, Option[Type], Expr)], exp2: Expr, loc: SourceLocation) extends Expr
case class Stm(exps: List[Expr], exp2: Expr, loc: SourceLocation) extends Expr
```

The refactoring is going to suck.

**Comment by magnus-madsen** [comment]
@mlutze @JonathanStarup CC

**Comment by mlutze** [comment]
What if we combined stm and let into one thing with an optional binding?

**Reply by JonathanStarup** [reply]
i like that yeah. localdef would also be in there
